### PR TITLE
Add Legend.title property and tests

### LINKED ISSUE #6769: legend title
I was wondering if an option can be added to bokeh to add titles to legends?
This is similar to a question that was asked here:
[Any way to create a legend title in Bokeh?](https://stackoverflow.com/questions/38155099/any-way-to-create-a-legend-title-in-bokeh
)

Something consistent with the bokeh syntax like:
```
p.legend.title = 'Legend Title'
```

### PR DESCRIPTION
**Pull request by canavandl**
- [x] issues: fixes #6769 
- [x] tests added / passed
- [ ] release document entry (if new feature or API change)

This PR:
* Adds a Legend.title property (and associated props) analogous to ColorBar.title
* Updates relevant examples and examples tests baseline

Screenshots:
Screen Shot 2019-05-02 at 7 44 21 PM
Screen Shot 2019-05-02 at 7 44 43 PM

### CODE REVIEW

File: bokeh/models/annotations.py [hunk 1]
```diff
@@ -183,6 +183,22 @@ class Legend(Annotation):
     when they are drawn.
     """)
 
+    title = String(help="""
```
Comment (canavandl) [inline bokeh/models/annotations.py]: see https://github.com/bokeh/bokeh/blob/master/bokeh/models/annotations.py#L318-L332 for reference

**Comment by bryevdv** [comment]
@canavandl The baselines are currently sensitive to chrome headless on linux, so for now we just have to accept those results as ground truth
```
[FAIL] BASELINE DOESN'T MATCH (make sure to update baselines before running tests):
[FAIL] diff --git a/tests/baselines/examples/models/file/legends b/tests/baselines/examples/models/file/legends
[FAIL] index 44462a0..c936b8d 100644
[FAIL] --- a/tests/baselines/examples/models/file/legends
[FAIL] +++ b/tests/baselines/examples/models/file/legends
[FAIL] @@ -1,9 +1,9 @@
[FAIL]  Plot bbox=[0, 0, 1000, 600]
[FAIL] -  Legend bbox=[151, 0, 698, 81]
[FAIL] -  LinearAxis bbox=[151, 495, 698, 24]
[FAIL] -  Legend bbox=[151, 519, 698, 81]
[FAIL] -  LinearAxis bbox=[120, 81, 31, 414]
[FAIL] -  Legend bbox=[0, 81, 120, 414]
[FAIL] -  LinearAxis bbox=[849, 81, 31, 414]
[FAIL] -  Legend bbox=[880, 81, 120, 414]
[FAIL] -  Title bbox=[151, 0, 698, 0]
[FAIL] +  Legend bbox=[152, 0, 696, 81]
[FAIL] +  LinearAxis bbox=[152, 495, 696, 24]
[FAIL] +  Legend bbox=[152, 519, 696, 81]
[FAIL] +  LinearAxis bbox=[121, 81, 31, 414]
[FAIL] +  Legend bbox=[0, 81, 121, 414]
[FAIL] +  LinearAxis bbox=[848, 81, 31, 414]
[FAIL] +  Legend bbox=[879, 81, 121, 414]
[FAIL] +  Title bbox=[152, 0, 696, 0]
```

**Comment by canavandl** [comment]
> @canavandl The baselines are currently sensitive to chrome headless on linux, so for now we just have to accept those results as ground truth
> 
> ```
> [FAIL] BASELINE DOESN'T MATCH (make sure to update baselines before running tests):
> [FAIL] diff --git a/tests/baselines/examples/models/file/legends b/tests/baselines/examples/models/file/legends
> [FAIL] index 44462a0..c936b8d 100644
> [FAIL] --- a/tests/baselines/examples/models/file/legends
> [FAIL] +++ b/tests/baselines/examples/models/file/legends
> [FAIL] @@ -1,9 +1,9 @@
> [FAIL]  Plot bbox=[0, 0, 1000, 600]
> [FAIL] -  Legend bbox=[151, 0, 698, 81]
> [FAIL] -  LinearAxis bbox=[151, 495, 698, 24]
> [FAIL] -  Legend bbox=[151, 519, 698, 81]
> [FAIL] -  LinearAxis bbox=[120, 81, 31, 414]
> [FAIL] -  Legend bbox=[0, 81, 120, 414]
> [FAIL] -  LinearAxis bbox=[849, 81, 31, 414]
> [FAIL] -  Legend bbox=[880, 81, 120, 414]
> [FAIL] -  Title bbox=[151, 0, 698, 0]
> [FAIL] +  Legend bbox=[152, 0, 696, 81]
> [FAIL] +  LinearAxis bbox=[152, 495, 696, 24]
> [FAIL] +  Legend bbox=[152, 519, 696, 81]
> [FAIL] +  LinearAxis bbox=[121, 81, 31, 414]
> [FAIL] +  Legend bbox=[0, 81, 121, 414]
> [FAIL] +  LinearAxis bbox=[848, 81, 31, 414]
> [FAIL] +  Legend bbox=[879, 81, 121, 414]
> [FAIL] +  Title bbox=[152, 0, 696, 0]
> ```

fixed

**Comment by mattpap** [comment]
Let's make sure that clickable legends still work with this (e.g. https://github.com/bokeh/bokeh/blob/13baccbb2835f7d5831c22b5cf68214e89c02618/sphinx/source/docs/user_guide/examples/interaction_legend_mute.py).

**Comment by bryevdv** [comment]
Good catch, this breaks the interaction. Clicking a legend entry activates the entry below. 

cc @canavandl 

Here is a complete test script:

```
import pandas as pd

from bokeh.palettes import Spectral4
from bokeh.plotting import figure, output_file, show
from bokeh.sampledata.stocks import AAPL, IBM, MSFT, GOOG

p = figure(plot_width=800, plot_height=250, x_axis_type="datetime")
p.title.text = 'Click on legend entries to mute the corresponding lines'

for data, name, color in zip([AAPL, IBM, MSFT, GOOG], ["AAPL", "IBM", "MSFT", "GOOG"], Spectral4):
    df = pd.DataFrame(data)
    df['date'] = pd.to_datetime(df['date'])
    p.line(df['date'], df['close'], line_width=2, color=color, alpha=0.8,
           muted_color=color, muted_alpha=0.2, legend=name)

p.legend.location = "top_left"
p.legend.click_policy="mute"
p.legend.title = 'Example Title'

output_file("interactive_legend.html", title="interactive_legend.py example")

show(p)
```

**Comment by canavandl** [comment]
> Good catch, this breaks the interaction. Clicking a legend entry activates the entry below.
> 
> cc @canavandl
> 
> Here is a complete test script:
> 
> ```
> import pandas as pd
> 
> from bokeh.palettes import Spectral4
> from bokeh.plotting import figure, output_file, show
> from bokeh.sampledata.stocks import AAPL, IBM, MSFT, GOOG
> 
> p = figure(plot_width=800, plot_height=250, x_axis_type="datetime")
> p.title.text = 'Click on legend entries to mute the corresponding lines'
> 
> for data, name, color in zip([AAPL, IBM, MSFT, GOOG], ["AAPL", "IBM", "MSFT", "GOOG"], Spectral4):
>     df = pd.DataFrame(data)
>     df['date'] = pd.to_datetime(df['date'])
>     p.line(df['date'], df['close'], line_width=2, color=color, alpha=0.8,
>            muted_color=color, muted_alpha=0.2, legend=name)
> 
> p.legend.location = "top_left"
> p.legend.click_policy="mute"
> p.legend.title = 'Example Title'
> 
> output_file("interactive_legend.html", title="interactive_legend.py example")
> 
> show(p)
> ```

fixed. Tested with both vertical and horizontal clickable legends.

**Comment by bryevdv** [comment]
I tried to add a selenium test but I could not get the glyph visibility to record changes in the test.

**Pull request merged**